app/src/sharedTest/java/org/oppia/android/app/parser/ListItemLeadingMarginSpanTest.kt

```diff
@@ -0,0 +1,620 @@
+      resourceBucketName,
+      entityType = "",
+      entityId = "",
+      imageCenterAlign = true,
+      displayLocale = appLanguageLocaleHandler.getDisplayLocale()
+    )
+    val htmlResult = activityScenarioRule.scenario.runWithActivity {
+      val textView: TextView = it.findViewById(R.id.test_list_content_text_view)
+      return@runWithActivity htmlParser.parseOppiaHtml(
+        "<p>You should know the following before going on:<br></p>" +
+          "<ol><li>The counting numbers (1, 2, 3, 4, 5 ….)</li>" +
+          "<li>How to tell whether one counting number is bigger or " +
+          "smaller than another</li></ol>",
+        textView
+      )
+    }
+
+    /* Reference: https://medium.com/androiddevelopers/spantastic-text-styling-with-spans-17b0c16b4568#e345 */
+    val bulletSpans =
+      htmlResult.getSpans(
+        0,
+        htmlResult.length,
+        ListItemLeadingMarginSpan.OlSpan::class.java
+      )
+    assertThat(bulletSpans.size.toLong()).isEqualTo(2)
+    val bulletSpan0 = bulletSpans[0] as ListItemLeadingMarginSpan.OlSpan
+    assertThat(bulletSpan0).isNotNull()
+
+    val leadingText = "1."
+    val expectedMargin = 2 * leadingText.length + spacingBeforeNumberedText
+
+    val bulletSpan0Margin = bulletSpan0.getLeadingMargin(true)
+    assertThat(bulletSpan0Margin).isEqualTo(expectedMargin)
+
+    val bulletSpan1 = bulletSpans[1] as ListItemLeadingMarginSpan.OlSpan
+    assertThat(bulletSpan1).isNotNull()
+  }
+
+  @Test
+  fun testDrawLeadingMargin_forNestedBulletItems_isdrawnCorrectlyWithIndentation() {
+    val htmlParser = htmlParserFactory.create(
+      resourceBucketName,
+      entityType = "",
+      entityId = "",
+      imageCenterAlign = true,
+      displayLocale = appLanguageLocaleHandler.getDisplayLocale()
+    )
+    val htmlResult = activityScenarioRule.scenario.runWithActivity {
+      val textView: TextView = it.findViewById(R.id.test_list_content_text_view)
+      return@runWithActivity htmlParser.parseOppiaHtml(
+        "<ul>" +
+          "        <li> Usage Data\", such as:" +
+          "          <ul>" +
+          "            <li>your answers to Lessons;</li>" +
+          "            <li>when you begin and end a Lesson;</li>" +
+          "          </ul>" +
+          "        </li>" +
+          "        <li> any contributions you make to the Site (such as feedback on" +
+          "            Lessons, edits to Lessons, and Lessons created);</li>" +
+          "      </ul>",
+        textView
+      )
+    }
+
+    /* Reference: https://medium.com/androiddevelopers/spantastic-text-styling-with-spans-17b0c16b4568#e345 */
+    val bulletSpans =
+      htmlResult.getSpans(
+        0,
+        htmlResult.length,
+        ListItemLeadingMarginSpan.UlSpan::class.java
+      )
+    assertThat(bulletSpans.size.toLong()).isEqualTo(4)
+    val bulletSpan0 = bulletSpans[0] as ListItemLeadingMarginSpan.UlSpan
+    assertThat(bulletSpan0).isNotNull()
```

Comment:
For this & other similar assertions below--are these actually needed?

Reply:
Not really. Removed it.